File: Leetcode/717. 1-bit and 2-bit Characters/solution.go
func isOneBitCharacter(bits []int) bool {
    i := 0
    result := false

    for i < len(bits) {
        if bits[i] == 0 {
            result = true;
            i++
            continue
        }

        result = false
        i += 2
    }

    return result
}

File: Leetcode/717. 1-bit and 2-bit Characters/solution.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut i = 0;
        let mut result = false;

        while i < bits.len() {
            if bits[i] == 0 {
                i += 1;
                result = true;
                continue;
            }

            i += 2;
            result = false;
        }

        result
    }
}